lib/everypolitician/country.rb

```diff
@@ -35,5 +35,21 @@ def legislature(query)
         query.all? { |k, v| l.__send__(k).to_s.downcase == v.to_s.downcase }
       end
     end
+
+    def upper_house
+      @upper_house ||= get_house('upper house')
+    end
+
+    def lower_house
+      @lower_house ||= get_house('lower house')
+    end
+
+    private
+
+    def get_house(type)
+      houses = legislatures.select { |l| l.type == type || l.type == 'unicameral legislature' }
+      return houses.first if houses.count == 1
+      houses.sort_by { |h| h.legislative_periods.max_by { |l| l[:start_date] }[:start_date] }.last
```

Comment:
This is a bit of a mouthful. It also seems odd to use `max_by` in the _inner_ block, rather than the outer one. If I'm reading it correctly, I think this would be a  simpler version:

```ruby
houses.max_by { |h| h.legislative_periods.map(&:start_date).max }
```

(Though I thought the `legislative_periods` were already guaranteed to be sorted. If not, they probably should be!)